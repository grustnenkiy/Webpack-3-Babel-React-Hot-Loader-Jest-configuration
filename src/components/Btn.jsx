import React from 'react';
import './btn.scss';

export class Btn extends React.Component {
    state = {
        isChanged: false
    }

    handleClick = () => {
        this.setState({ isChanged: !this.state.isChanged })
    }

    render() {
        return (
            <div className="btn">
                <button onClick={this.handleClick} className="btn-button">Change</button>
                <span className={(this.state.isChanged ? "btn-text" : "btn-text-highlighted")}>Very important text</span>
            </div>
        )
    }
}