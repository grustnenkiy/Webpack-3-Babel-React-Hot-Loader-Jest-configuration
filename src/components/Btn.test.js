import React from 'react';
import { Btn } from './Btn';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
    const btn = renderer
        .create(<Btn />)
        .toJSON();
    expect(btn).toMatchSnapshot();
});