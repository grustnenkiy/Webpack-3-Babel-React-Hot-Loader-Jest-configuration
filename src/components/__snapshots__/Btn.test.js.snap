// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="btn"
>
  <button
    className="btn-button"
    onClick={[Function]}
  >
    Change
  </button>
  <span
    className="btn-text-highlighted"
  >
    Very important text
  </span>
</div>
`;
